### PR TITLE
adding nping check after performance test

### PR DESCRIPTION
**Pull request by vaishnavibhat**
doing a ping check with tcp packets after iperf test.

Signed-off-by: Vaishnavi Bhat <vaishnavi@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by vaishnavibhat** [comment]
avocado run iperf_test.py -m iperf_test.py.data/iperf_test.yaml 
/usr/local/lib/python3.6/site-packages/avocado_framework-80.0-py3.6.egg/avocado/plugins/run.py:285: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp. 
  FutureWarning)
JOB ID     : a304025d692a34ba6d1d6742750a4b3070d6f06f
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2020-09-09T15.20-a304025/job.log
 (1/9) iperf_test.py:Iperf.test;1500-9990: PASS (83.68 s)
 (2/9) iperf_test.py:Iperf.test;2000-df88: PASS (82.70 s)
 (3/9) iperf_test.py:Iperf.test;3000-769a: PASS (82.65 s)
 (4/9) iperf_test.py:Iperf.test;4000-548c: PASS (82.67 s)
 (5/9) iperf_test.py:Iperf.test;5000-9a06: PASS (82.70 s)
 (6/9) iperf_test.py:Iperf.test;6000-1b19: PASS (82.63 s)
 (7/9) iperf_test.py:Iperf.test;7000-c2d9: PASS (82.67 s)
 (8/9) iperf_test.py:Iperf.test;8000-73a3: PASS (82.60 s)
 (9/9) iperf_test.py:Iperf.test;9000-74f9: PASS (82.70 s)
RESULTS    : PASS 9 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2020-09-09T15.20-a304025/results.html
JOB TIME   : 745.28 s

[iperf.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/5194434/iperf.log)

File: io/net/iperf_test.py [hunk 1]
```diff
@@ -98,10 +99,12 @@ def setUp(self):
         archive.extract(tarball, self.iperf)
         self.version = os.path.basename(tarball.split('.tar')[0])
         self.iperf_dir = os.path.join(self.iperf, self.version)
-        destination = "%s:/tmp" % self.peer_ip
-        output = self.session.copy_files(self.iperf_dir, destination,
-                                         recursive=True)
-        if not output:
+        out = self.session.get_raw_ssh_command('ls')
```
Comment (smruti77) [inline io/net/iperf_test.py:102]: @vaishnavibhat why you changed this like before .? already handle this as per utils.

**Review by smruti77 — CHANGES_REQUESTED** [review]

File: io/net/iperf_test.py [hunk 2]
```diff
@@ -157,10 +177,7 @@ def tearDown(self):
         except Exception:
             self.peer_public_networkinterface.set_mtu('1500')
         self.networkinterface.remove_ipaddr(self.ipaddr, self.netmask)
-        try:
-            self.networkinterface.restore_from_backup()
-        except Exception:
-            self.log.info("backup file not availbale, could not restore file.")
+        self.networkinterface.restore_from_backup()
```
Comment (smruti77) [inline io/net/iperf_test.py:180]: @vaishnavibhat this is also same, don't remove try ..except. we handle this as we are getting exception if backup file was not there.

**Review by smruti77 — CHANGES_REQUESTED** [review]

File: io/net/iperf_test.py [hunk 3]
```diff
@@ -140,6 +144,22 @@ def test(self):
                               % ((tput*100)/speed, self.expected_tp,
                                  str(tput)+'Mb/sec'))
 
+    def nping(self):
+        """
+        perform nping
+        """
+        detected_distro = distro.detect()
+        if detected_distro.name == "SuSE":
+            cmd = "./nping/nping --tcp %s -c 10" % self.peer_ip
```
Comment (smruti77) [inline io/net/iperf_test.py:153]: @vaishnavibhat nmap things need to install before for suse.

**Review by narasimhan-v — CHANGES_REQUESTED** [review]
@vaishnavibhat Can you please explain more in the commit message ?

**Comment by PraveenPenguin** [comment]
@vaishnavibhat if you are still perusing this PR , please feel Free to open PR having elaborative commit message , as this PR also has conflict

**Pull request closed without merging**